Functions/Quantification/xASL_quant_M0.m

```diff
@@ -364,31 +361,40 @@
     end    
     
     %% Create the figure
-    % SignalPercentage
-    subplot(2, 2, 2);
-    plot(1:size(M0IM,3), SignalPercentage,'k');
-    title('Signal left after background suppression');
-    ylabel('Residual signal (%)');
-    xlabel('Slice (integer)');    
-    % Before correction
-    subplot(2, 2, 3);
-    plot(1:size(M0IM,3), SignalBeforeCorrection);
-    title('Before correction');
-    ylabel('Pseudo-M0 (a.u.)');
-    xlabel('Slice (integer)');
-    % After correction
-    subplot(2, 2, 4);
-    plot(1:size(M0IM,3), SignalAfterCorrection,'r');
-    title('After correction');
-    ylabel('Pseudo-M0 (a.u.)');
-    xlabel('Slice (integer)');
-
-    % Save and close the figure
-    SavePath = fullfile(x.D.M0CheckDir, 'RevertBsupFxControl.jpg');
-	xASL_adm_CreateDir(x.D.M0CheckDir);
-    saveas(FigureHandle, SavePath, 'jpg');
-    close(FigureHandle);
-    
+	% Only if java virtual machine is used
+	if usejava('desktop') || usejava('jvm')
+		% Initialize the figure
+		FigureHandle = figure('visible','off');
+		subplot(2, 2, 1);
+
+		% SignalPercentage
+		subplot(2, 2, 2);
+		plot(1:size(M0IM,3), SignalPercentage,'k');
+		title('Signal left after background suppression');
+		ylabel('Residual signal (%)');
+		xlabel('Slice (integer)');
+		% Before correction
+		subplot(2, 2, 3);
+		plot(1:size(M0IM,3), SignalBeforeCorrection);
+		title('Before correction');
+		ylabel('Pseudo-M0 (a.u.)');
+		xlabel('Slice (integer)');
+		% After correction
+		subplot(2, 2, 4);
+		plot(1:size(M0IM,3), SignalAfterCorrection,'r');
+		title('After correction');
+		ylabel('Pseudo-M0 (a.u.)');
+		xlabel('Slice (integer)');
+
+		% Save and close the figure
+		SavePath = fullfile(x.D.M0CheckDir, 'RevertBsupFxControl.jpg');

```

Comment:
Actually, this is probably a bug somebody introduced before (not me ;)

This saves a single jpg that will be overwritten for each ASL subject/session right?
So it should probably be [x.SUBJECT '_' x.SESSION 'RevertBsupFxControl.jpg' ?
Or don't you want to check this for each subject (then still this will only show for the last subject/session)?

Reply:
Good point! Actually, this figure depends on the sequence parameters, so it won't really matter if this differs. But in the future it could, so I'm adding the session+name.